pilot/pkg/networking/ambientgen/ztunnelgen.go

```diff
@@ -1089,50 +1090,55 @@ func (g *ZTunnelConfigGenerator) buildInboundCaptureListener(proxy *model.Proxy,
-							RouteConfig: &route.RouteConfiguration{
-								Name: "local_route",
-								VirtualHosts: []*route.VirtualHost{{
-									Name:    "local_service",
-									Domains: []string{"*"},
-									Routes: []*route.Route{{
-										Match: &route.RouteMatch{PathSpecifier: &route.RouteMatch_ConnectMatcher_{
-											ConnectMatcher: &route.RouteMatch_ConnectMatcher{},
+
+			inboundHCM := &httpconn.HttpConnectionManager{
+				AccessLog:  accessLogString("inbound hcm"),
+				CodecType:  0,
+				StatPrefix: "inbound_hcm",
+				RouteSpecifier: &httpconn.HttpConnectionManager_RouteConfig{
+					RouteConfig: &route.RouteConfiguration{
+						Name: "local_route",
+						VirtualHosts: []*route.VirtualHost{{
+							Name:    "local_service",
+							Domains: []string{"*"},
+							Routes: []*route.Route{{
+								Match: &route.RouteMatch{PathSpecifier: &route.RouteMatch_ConnectMatcher_{
+									ConnectMatcher: &route.RouteMatch_ConnectMatcher{},
+								}},
+								Action: &route.Route_Route{
+									Route: &route.RouteAction{
+										UpgradeConfigs: []*route.RouteAction_UpgradeConfig{{
+											UpgradeType:   "CONNECT",
+											ConnectConfig: &route.RouteAction_UpgradeConfig_ConnectConfig{},
 										}},
-										Action: &route.Route_Route{
-											Route: &route.RouteAction{
-												UpgradeConfigs: []*route.RouteAction_UpgradeConfig{{
-													UpgradeType:   "CONNECT",
-													ConnectConfig: &route.RouteAction_UpgradeConfig_ConnectConfig{},
-												}},
-												ClusterSpecifier: &route.RouteAction_Cluster{
-													Cluster: "virtual_inbound",
-												},
-											},
+										ClusterSpecifier: &route.RouteAction_Cluster{
+											Cluster: "virtual_inbound",
 										},
-									}},
-								}},
-							},
-						},
-						// TODO rewrite destination port to original_dest port
-						HttpFilters: []*httpconn.HttpFilter{{
-							Name:       "envoy.filters.http.router",
-							ConfigType: &httpconn.HttpFilter_TypedConfig{TypedConfig: protoconv.MessageToAny(&routerfilter.Router{})},
-						}},
-						Http2ProtocolOptions: &core.Http2ProtocolOptions{
-							AllowConnect: true,
-						},
-						UpgradeConfigs: []*httpconn.HttpConnectionManager_UpgradeConfig{{
-							UpgradeType: "CONNECT",
+									},
+								},
+							}},
 						}},
-					}),
+					},
+				},
+				// TODO rewrite destination port to original_dest port
+				HttpFilters: []*httpconn.HttpFilter{{
+					Name:       "envoy.filters.http.router",
+					ConfigType: &httpconn.HttpFilter_TypedConfig{TypedConfig: protoconv.MessageToAny(&routerfilter.Router{})},
+				}},
+				Http2ProtocolOptions: &core.Http2ProtocolOptions{
+					AllowConnect: true,
+				},
+				UpgradeConfigs: []*httpconn.HttpConnectionManager_UpgradeConfig{{
+					UpgradeType: "CONNECT",
+				}},
+			}
+
+			v1alpha3.ConfigureTracing(push, proxy, inboundHCM, networking.ListenerClassSidecarInbound)
```

Comment:
I am not sure we want tracing in ztunnel since its only doing L4 and we want to keep a minimal feature surface area.

I don't mind adding it here if we want, but with the expectation it will probably be dropped when we move to https://github.com/istio/ztunnel

Reply:
http traffic will ztunnel -> waypoint -> ztunnel, tracing will helpful if there's network issue between ztunnel and waypoint?
correct me if I'm wrong.